Tone/instrument/CymbalSynth.js

```diff
@@ -0,0 +1,304 @@
+
+	/**
+	 *  the default values
+	 *  @static
+	 *  @const
+	 *  @type {Object}
+	 */
+	Tone.CymbalSynth.defaults = {
+		"harmonicity" : 0,
+		"frequency" : 200,
+		"strike" : {
+			"volume" : 0,
+			"cutoff" : 3500, 
+			"resonance" : 6,
+			"envelope": { 
+				"attack" : 0.01, 
+				"decay" : 0.25
+			}
+		},
+		"body" : {
+			"volume" : 0,
+			"cutoff" : 7000,
+			"resonance" : 6,
+			"envelope" : {
+				"attack" : 0.01,
+				"decay" : 1.5
+			}
+
+
+		},
+	};
+
+	/**
+	 *  Trigger the cymbal at a given time
+	 *  @param  {Time} [time=now]     the time, if not given is now
+	 *  @param  {Number} [velocity = 1] velocity defaults to 1
+	 *  @returns {Tone.Drumsynth} this
+	 *  @example
+	 *  cymbal.triggerAttack("0:1:1". 0.75);
+	 */
+	Tone.CymbalSynth.prototype.triggerAttack = function(time, velocity){
+		time = this.toSeconds(time);
+		this.body.envelope.triggerAttack(time, velocity);
+		this.strike.envelope.triggerAttack(time, velocity);
+		return this;
+	};
+
+	/**
+	 *  Trigger the release of the cymbal.
+	 *  Can be used to [choke](https://en.wikipedia.org/wiki/Cymbal_choke) 
+	 *  the cymbal
+	 *  @param  {Time} [time=now] the time the cymbal will release
+	 *  @returns {Tone.CymbalSynth} this
+	 *  @example
+	 *  //set the release to a small amount for a realistic choke
+	 *  cymbal.body.envelope.release = 0.2;
+	 *  cymbal.triggerAttack("0:0:0");
+	 *  cymbal.triggerRelease("0:0:0" + 0.4);
+	 */
+	Tone.CymbalSynth.prototype.triggerRelease = function(time){
+		this.strike.envelope.triggerRelease(time);
+		this.body.envelope.triggerRelease(time);
+		return this;
+	};
+
+	/**
+	 * Clean up.
+	 * @return {Tone.CymbalSynth} this
+	 */
+	Tone.CymbalSynth.prototype.dispose = function(){
+		Tone.Instrument.prototype.dispose.call(this);
+		this.harmonicity.dispose();
+		this.harmonicity = null;
+		this.frequency.dispose();
+		this.frequency = null;
+		this.inharmRatios = null;
+		this.harmRatios = null;
+
+		for(var i = 0; i <6; i++){
+			this._oscillators[i].dispose();
+			this._scaledSignals[i].dispose();
+			this._freqMult[i].dispose();
+		}
+		this._oscillators = null;
+		this._scaledSignals = null;
+		this._freqMult = null;
+
+		this.body.dispose();
+		this.strike.dispose();
+		this.strike = null;
+		this.body = null;
+		return this;
+	};
+
+	/**
+	 * Cymbal part helper class
+	 * @private
+	 */
+	var CymbalComponent = function(options){
+		this.output = this.context.createGain();
+		this.volume = new Tone.Signal({
+			"param" : this.output.gain, 
+			"units" : Tone.Type.Decibels,
+			"value" : options.volume
+		});
+		this._readOnly(["volume"]);
+		this._highPass = new Tone.Filter({
+			"type" : "highpass"
+		}).connect(this.output);
+		this.envelope = new Tone.AmplitudeEnvelope({
+			"attack" : options.envelope.attack,
+			"decay" : options.envelope.decay,
+			"sustain" : 0,
+			"release": 0,
+			"attackCurve" : "exponential"
+
+		}).connect(this._highPass);
+		this.input = new Tone.Filter({
```

Comment:
the reason I made this private is because there is no need to access any of its attributes directly. the highpass filter frequency and q are both modified via scalars on the main frequency and resonance values. so i don't really want people changing these since there is not much to change (perhaps the rolloff but thats it)